episodes/43-software-release.md

```diff
@@ -19,286 +19,288 @@ exercises: 20
-```
+## What is a Version Number Anyway?
 
-### Setting up our Poetry Config
+Software version numbers are everywhere,
+and there are many different ways to do it.
+A popular one to consider is [**Semantic Versioning**](https://semver.org/),
+where a given version number uses the format MAJOR.MINOR.PATCH.
+You increment the:
 
-Poetry uses a **pyproject.toml** file to describe
-the build system and requirements of the distributable package.
-This file format was introduced to solve problems with bootstrapping packages
-(the processing we do to prepare to process something)
-using the older convention with **setup.py** files and to support a wider range of build tools.
-It is described in
-[PEP 518 (Specifying Minimum Build System Requirements for Python Projects)](https://www.python.org/dev/peps/pep-0518/).
+- MAJOR version when you make incompatible API changes
+- MINOR version when you add functionality in a backwards compatible manner
+- PATCH version when you make backwards compatible bug fixes
 
-Make sure you are in the root directory of your software project
-and have activated your virtual environment,
-then we are ready to begin.
+You can also add a hyphen followed by characters to denote a pre-release version,
+e.g. 1.0.0-alpha1 (first alpha release) or 1.2.3-beta4 (fourth beta release)
 
-To create a `pyproject.toml` file for our code, we can use `poetry init`.
-This will guide us through the most important settings -
-for each prompt, we either enter our data or accept the default.
 
-*Displayed below are the questions you should see
-with the recommended responses to each question so try to follow these,
-although use your own contact details!*
+::::::::::::::::::::::::::::::::::::::::::::::::::
 
-**NB: When you get to the questions about defining our dependencies,
-answer no, so we can do this separately later.**
+## Tagging a Release in GitHub
 
-:::::::::::::::::::::::::::::::::::::::::  callout
+There are many ways in which Git and GitHub can help us make a software release from our code.
+For example, we can use GitHub website to create a new release. 
+In this episode, we will look at how to do this using Git **tagging** at the command line.
 
-## It's Not Interactive?
+Let us see what tags we currently have in our repository:
 
-If you're using Git Bash for Windows, depending on your configuration, you may find after typing this command that you don't have an interactive set of questions displayed.
-Instead, you may find the `pyproject.toml` file is simply generated with a set of default values.
+```bash
+$ git tag
+```
 
-If this happens, you can edit the `pyproject.toml` file and change the values in this file, similarly to how we have in the output below.
+Since we have not tagged any commits yet, there is unsurprisingly no output.
+We can create a new tag on the last commit in our `main` branch by doing:
 
-:::::::::::::::::::::::::::::::::::::::::
+```bash
+$ git tag -a 1.0.0 -m "Version 1.0.0"
```

Comment:
```suggestion
$ git tag -a 0.1.0 -m "Version 0.1.0"
```
Changing the tag to 0.1.0 since that's the version number set in `pyproject.toml`